pkg/chains/storage/oci/oci_test.go

```diff
@@ -70,14 +68,24 @@ func TestBackend_StorePayload(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			b := &Backend{
 				logger: logger,
-				tr:     tt.fields.tr,
-				cfg:    tt.fields.cfg,
-				kc:     tt.fields.kc,
-				auth:   tt.fields.auth,
 			}
-			if err := b.StorePayload(ctx, tt.args.rawPayload, tt.args.signature, tt.args.storageOpts); (err != nil) != tt.wantErr {
+
+			cleanup := setupMocks()
+			defer cleanup()
+
+			if err := b.StorePayload(ctx, tt.fields.tr, tt.args.rawPayload, tt.args.signature, tt.args.storageOpts); (err != nil) != tt.wantErr {
 				t.Errorf("Backend.StorePayload() error = %v, wantErr %v", err, tt.wantErr)
 			}
 		})
 	}
 }
+
+func setupMocks() func() {
+	oldAuth := getAuthenticator
+	getAuthenticator = func(_ context.Context, _ *v1beta1.TaskRun, _ kubernetes.Interface) (remote.Option, error) {
```

Comment:
Can we setup this per-backend instead of at the package level?

Reply:
When we initialize a k8s Keychain to override default one, we need namespace and service account specified in taskrun spec. And different taskrun might have different spec for namespace and service account to use. 
So IIUC we might want to setup this for per taskrun? Please correct me if I'm wrong.

Reply:
Yup, but here we're overriding a package level func, which will affect all OCI backends (and worst case if tests run in parallel this may cause issues).

Similar to how you're defining clients per backend now, you can do the same for the authenticator func e.g.

```go
type Backend struct {
  ...
  getAuthenticator func(context.Context, *v1beta1.TaskRun, kubernetes.Interface) (remote.Option, error)
}
```

and then within tests you can just set the func to the no-op implementation.